mono/mini/interp/interp.c

```diff
@@ -173,8 +173,8 @@ debug_enter (MonoInvocation *frame, int *tracing)
 		g_print  ("%s)\n", args);
 		g_free (args);
 	}
-	if (mono_profiler_events & MONO_PROFILE_ENTER_LEAVE)
-		mono_profiler_method_enter (frame->runtime_method->method);
+	if (mono_profiler_should_instrument_method (frame->runtime_method->method, TRUE))
```

Comment:
That's implicit. If no filter functions are installed, enter/leave is not enabled. I don't think requiring it to be explicitly enabled adds much value compared to just installing a filter function if you want to use the feature.